datajoint/connection.py

```diff
@@ -122,6 +122,8 @@ def query(self, query, args=(), as_dict=False):
                     be violated. You can switch off this behavior by setting the 'database.reconnect' to False.
                     ''')
                 self.connect()
+                logger.debug("Re-Executing SQL:" + query[0:300])
```

Comment:
I think the default setting should be to disable reconnects.  This is a compromise we have sometimes made but automatic reconnects may have deleterious effects.